src/MarieAssembler.c

```diff
@@ -1010,66 +996,51 @@ wchar_t *LoadFileIntoMemory(wchar_t *FileName, int *Success) {
 	return Result;
 }
 
-int ApplicationMain(int ArgCount, wchar_t **Args) {
-	if (!(ArgCount == 2 || ArgCount == 3)) {
-		const wchar_t *HelpMessage = 
-			L"* Usage: %s <Path To Source> [Output File Name]\n"
-			"* This program compiles Marie Assembly code into logisim rom images.\n"
-			"* <Path To Source> is the path to the assembly code you'd like to assemble.\n"
-			"* [Output File Name] is the name of the files that this program will emit.\n"
-			"** If [Output File Name] is not provied, then this program generates [Output File Name] in the form of OUT_<File Name>.\n"
-			"* For Every compliation:\n"
-			"** A Raw Image file will be emitted as [Output File Name].hex\n"
-			"** A listing file will be emitted as [Output File Name].lst\n"
-			"** A symbol table will be output as [Output File Name].sym\n"
-			"** A logisim image file will bit output as [Output File Name].LogisimImage\n";
-		wprintf(HelpMessage, Args[0]);
-		return 0;
-	}
-
+int ApplicationMain(FILE *InFile, int InFileSize, FILE *OutLogisim, FILE *OutRawHex, FILE *OutSymbolTable, FILE *OutListing) {
 	int Success = TRUE;
+	if (InFile == 0) {
+		Success = FALSE;
+		printf("A input file was not provided!\n");
+	}
 
-	file_state FileState = {
-		.Line = 1,
-		.Column = 0,
-	};
-	FileState.At = LoadFileIntoMemory(Args[1], &Success);
-	if (Success == FALSE) { return 0; }
+	if (Success) {
+		file_state FileState = {
+			.Line = 1,
+			.Column = 0,
+		};
+		FileState.At = LoadFileIntoMemory(InFile, InFileSize, &Success);
 
-	paged_list *IdentifierDestinationList = AllocatePagedList(sizeof(identifier_dest), 10);
-	paged_list *IdentifierSourceList = AllocatePagedList(sizeof(identifier_source), 10);
-	
-	if (Assemble(&FileState, IdentifierDestinationList, IdentifierSourceList)) {
-		wchar_t *OutputFileName;
-		int OutputNameLength;
+		paged_list *IdentifierDestinationList = AllocatePagedList(sizeof(identifier_dest), 10);
+		paged_list *IdentifierSourceList = AllocatePagedList(sizeof(identifier_source), 10);
+		
+		Success = Assemble(&FileState, IdentifierDestinationList, IdentifierSourceList);
+		
+		if (Success) {
+			if ((OutLogisim == 0) && (OutRawHex == 0) && (OutSymbolTable == 0) && (OutListing == 0)) {
```

Comment:
Good job with the args, but this is going a little overboard. At the very least we should be spitting out the raw hex or assembly instructions when we call this program.

Reply:
Personally, I think I drew the line at the right spot. We should only output what we are asked. We don't have any way of knowing what the user will want if they don't specify an output.

Reply:
This is a MARIE lang -> Assembly lang compiler right? It'd make sense at a minimum to make that conversion by default. Ex. I wouldn't specify gcc to compile my C code to binaries, that'd just be the intention of using the program by default.

Reply:
This takes MARIE assembly lang and outputs MARIE machine code.

I feel apprehensive about this because we have two different outputs for MARIE machine code that the user might want: The naked hex, and a Logisim rom image. I don't feel like there's a safe assumption to make here.